source/state_representation/src/parameters/ParameterMap.cpp

```diff
@@ -34,22 +34,25 @@ void ParameterMap::set_parameter(const std::shared_ptr<ParameterInterface>& para
 }
 
 void ParameterMap::set_parameters(const ParameterInterfaceList& parameters) {
-  for (const auto& param: parameters) {
+  for (const auto& param : parameters) {
     this->set_parameter(param);
   }
 }
 
 void ParameterMap::set_parameters(const ParameterInterfaceMap& parameters) {
-  for (const auto& param_it: parameters) {
+  for (const auto& param_it : parameters) {
     this->set_parameter(param_it.second);
   }
 }
 
 void ParameterMap::assert_parameter_valid(const std::shared_ptr<ParameterInterface>& parameter) {
-  if (this->parameters_.at(parameter->get_name())->get_type() != parameter->get_type()) {
-    throw exceptions::InvalidParameterException(
-        "Parameter '" + parameter->get_name() + "' exists, but has unexpected type."
-    );
+  try {
+    if (this->parameters_.at(parameter->get_name())->get_parameter_type() != parameter->get_parameter_type()) {
```

Comment:
Thank you for making me do this haha, the current implementation really didn't do anything because the type (StateType) of two ParameterInterfaces will always be the same! We have to compare the parameter type instead,

And we even have to add a special condition for when the parameter type is state to check that the `parameter_state_type` corresponds, no?

Reply:
haha well that makes a lot of sense, hard to catch though!